be/src/Unic.UrlMapper2/code/ContentSearch/ComputedFields/UrlMapperComputedFieldBase.cs

```diff
@@ -0,0 +1,48 @@
+﻿namespace Unic.UrlMapper2.ContentSearch.ComputedFields
+{
+    using Sitecore.ContentSearch;
+    using Sitecore.ContentSearch.ComputedFields;
+    using Sitecore.Data.Items;
+    using Unic.UrlMapper2.DependencyInjection;
+    using Unic.UrlMapper2.Services;
+
+    public abstract class UrlMapperComputedFieldBase : IComputedIndexField
+    {
+        private Scope scope;
+
+        public string FieldName { get; set; }
+        public string ReturnType { get; set; }
+
+        public object ComputeFieldValue(IIndexable indexable)
+        {
+            // ReSharper disable once InvertIf
+            if (this.ShouldCompute(indexable))
+            {
+                object result;
+
+                using (this.scope = Container.CreateScope())
+                {
+                    result = this.Compute(indexable);
+                }
+
+                this.scope = null;
```

Comment:
I'm not sure if we still need the concept of a scope safe computed field. This was necessary when we used to have SimpleInjector as a DI container. SimpleInjector would throw an exception when you were resolving a scoped service in a context outside of an `HttpRequest`. Microsoft DI, which we use now, does handle it differently. It will just resolve the service with a transient lifetime. Nevertheless, I does not hurt the implementation.

Reply:
Yeah, I checked the behaviour and while there was no exception being generated, the service was resolved as transient while being registered as scoped. As this is rather unexpected in this context, i decided to introduce the scope safe implementation.